src/interp.rs

```diff
@@ -245,6 +252,37 @@ impl<'a, 'tcx> Machine<'a, 'tcx> {
                     _ => unimplemented!()
                 }
             },
+            _ => unimplemented!("{:?}", rvalue)
+        }
+    }
+
+    fn binary_op(&self,
+                 binop: BinOp,
+                 lhs: TyVal<'tcx>,
+                 rhs: TyVal<'tcx>)
+                 -> Value {
+        match binop {
+            BinOp::Add => {
+                match (lhs.val, rhs.val) {
+                    (Value::Int(lv), Value::Int(rv)) => {
+                        assert_eq!(lhs.ty, rhs.ty);
```

Comment:
I assume `typ` is a type field of sorts, but they're both `Value::Int` already?

Reply:
That's true. But `Value::Int` can be any int type, (i8, i16, i32... etc) so I still need to check that the native Rust types match.